JASP-Engine/JASP/R/ttestindependentsamples.R

```diff
@@ -778,80 +765,236 @@ TTestIndependentSamples <- function(dataset = NULL, options, perform = "run",
+																	na.rm = TRUE, .drop = FALSE)
+
+				colnames(summaryStat)[which(colnames(summaryStat) == .v(variables[variableIndex]))] <- "dependent"
+				colnames(summaryStat)[which(colnames(summaryStat) == .v(groups))] <- "groupingVariable"
+
+				pd <- ggplot2::position_dodge(0.2)
+
+				p <- ggplot2::ggplot(summaryStat, ggplot2::aes(x = groupingVariable,
+																											 y = dependent, group = 1)) + ggplot2::geom_errorbar(ggplot2::aes(ymin = ciLower,
+																																																												ymax = ciUpper), colour = "black", width = 0.2, position = pd) +
+					ggplot2::geom_line(position = pd, size = 0.7) + ggplot2::geom_point(position = pd,
+																																							size = 4) + ggplot2::ylab(unlist(options$variables[variableIndex])) + ggplot2::xlab(options$groupingVariable) +
+					base_breaks_y(summaryStat) + base_breaks_x(summaryStat$groupingVariable)
+
+				p <- JASPgraphs::themeJasp(p)
+
+				imgObj <- .writeImage(width = options$plotWidth,
+															height = options$plotHeight,
+															plot = p)
+
+				descriptivesPlot[["data"]] <- imgObj[["png"]]
+				descriptivesPlot[["obj"]] <- imgObj[["obj"]]
+
+			}
+
 			descriptivesPlot[["convertible"]] <- TRUE
 			descriptivesPlot[["status"]] <- "complete"
-			
+
 			descriptivesPlotList[[variableIndex]] <- descriptivesPlot
 
 		}
-		
+
 		return(descriptivesPlotList)
 
 	} else {
-	    
-	    return(NULL)
-	    
+
+		return(NULL)
+
+	}
+
+}
+
+
+confidenceLimitsEffectSizes <- function(ncp, df, conf.level=.95, alpha.lower=NULL, alpha.upper=NULL, t.value, tol=1e-9, sup.int.warns=TRUE, ...)
+{
+	# This function comes from the MBESS package, version 4.6, by Ken Kelley
+	# https://cran.r-project.org/web/packages/MBESS/index.html
+	# Note this function is new in version 4, replacing what was used in prior versions.
+	# Internal functions for the noncentral t distribution; two appraoches.
+	###########
+
+	if(missing(ncp))
+	{
+		if(missing(t.value)) stop("You need to specify either 'ncp' or its alias, 't.value,' you have not specified either")
+		ncp <- t.value
+	}
+
+	# General stop checks.
+	if(df <= 0) stop("The degrees of freedom must be some positive value.", call.=FALSE)
+
+	if(abs(ncp) > 37.62) print("The observed noncentrality parameter of the noncentral t-distribution has exceeded 37.62 in magnitude (R's limitation for accurate probabilities from the noncentral t-distribution) in the function's iterative search for the appropriate value(s). The results may be fine, but they might be inaccurate; use caution.")
+
+	if(sup.int.warns==TRUE) Orig.warn <- options()$warn; options(warn=-1)
+
+	if(!is.null(conf.level) & is.null(alpha.lower) & !is.null(alpha.upper)) stop("You must choose either to use \'conf.level\' or define the \'lower.alpha\' and \'upper.alpha\' values; here, \'upper.alpha\' is specified but \'lower.alpha\' is not", call.=FALSE)
```

Comment:
The message confused me. Perhaps the following could do:

stop("You must either use \'conf.level\', or define the \'lower.alpha\' and \'upper.alpha\' values; here, \'upper.alpha\' is specified but \'lower.alpha\' is not", call.=FALSE)